epics/ca.py

```diff
@@ -1031,13 +1031,15 @@ def array_cast(data, count, ntype, use_numpy):
             else:
                 out = numpy.ctypeslib.as_array(memcopy(data))
         else:
-            out = memcopy(data)
+            out = list(data)
         return out
 
-    def unpack(data, count, ntype, use_numpy):
+    def unpack(data, count, ntype, use_numpy, elem_count):
         "simple, native data type"
         if data is None:
             return None
+        elif ntype == dbr.CHAR and elem_count > 1:
```

Comment:
Is count == 1 in that case? 

If count > 1, it's an array for sure. If count==1 it *might* be an array.